packages/date-picker/src/vaadin-month-calendar-mixin.js

```diff
@@ -334,10 +334,70 @@ export const MonthCalendarMixin = (superClass) =>
 
     /** @private */
     _showWeekNumbersChanged(showWeekNumbers, i18n) {
-      if (showWeekNumbers && i18n && i18n.firstDayOfWeek === 1) {
+      if (this.__computeShowWeekSeparator(showWeekNumbers, i18n)) {
         this.setAttribute('week-numbers', '');
       } else {
         this.removeAttribute('week-numbers');
       }
     }
+
+    // eslint-disable-next-line @typescript-eslint/max-params
+    __computeDatePart(date, focusedDate, selectedDate, minDate, maxDate, isDateDisabled, enteredDate, hasFocus) {
+      const result = ['date'];
+
+      if (this.__isDayDisabled(date, minDate, maxDate, isDateDisabled)) {
+        result.push('disabled');
+      }
+
+      if (dateEquals(date, focusedDate) && (hasFocus || dateEquals(date, enteredDate))) {
+        result.push('focused');
+      }
+
+      if (this.__isDaySelected(date, selectedDate)) {
+        result.push('selected');
+      }
+
+      if (this._isToday(date)) {
+        result.push('today');
+      }
+
+      if (date < normalizeDate(new Date())) {
+        result.push('past');
+      }
+
+      if (date > normalizeDate(new Date())) {
+        result.push('future');
+      }
+
+      return result.join(' ');
+    }
+
+    /** @private */
+    __isDaySelected(date, selectedDate) {
+      return dateEquals(date, selectedDate);
+    }
+
+    /** @private */
```

Comment:
nit: we could consider marking computed methods as protected since they are used by the template.

Reply:
I chose to keep them private to prevent the temptation of using them outside the component. :)